# COVID-19 Related Queries for BTE Use Case Exploration 

# TABLE OF CONTENTS

## 1 TMPRSS2 Related Queries
###  &emsp; 1.1 TMPRSS2 -> NAFAMOSTAT
####  &emsp; &emsp; 1.1.1 TMPRSS2 -> NAFAMOSTAT Query
####  &emsp; &emsp; 1.1.2 TMPRSS2 -> NAFAMOSTAT Results table 
###  &emsp; 1.2 Question: Is the only way to get from TMPRSS2 to Blood Coagulation through Nafamostat (Fut 175)?
#### &emsp;&emsp; 1.2.1 TMPRSS2 to Blood Coagulation through any node type
#### &emsp;&emsp; 1.2.2 TMPRSS2 to Blood Coagulation through any ChemicalSubstance type node
### &emsp; 1.3 TMPRSS2 -> Camostat
#### &emsp;&emsp; 1.3.1 In what ways can the TMPRSS2 Gene be connected to Camostat?
## 2 CAMOSTAT Related Queries
### &emsp; 2.1 Camostat to Protease Activity 
#### &emsp;&emsp;  2.1.1 Camostat to Protease Activity Query 
#### &emsp;&emsp; 2.1.2 Camostat to Protease Activity Results
### &emsp; 2.2 Camostat to Protease Related Genes
#### &emsp;&emsp; 2.2.1 Camostat to Protease Related Genes Query
#### &emsp;&emsp; 2.2.2 Camostat to Protease Related Genes Results
### &emsp; 2.3 Camostat to Coagulation
#### &emsp;&emsp; 2.3.1 Camostat to Coagulation Results


# 1 TMPRSS2 Related Queries¶

In [8]:
## First get all the functions set up
import pandas as pd
## Load BTE
from biothings_explorer.user_query_dispatcher import FindConnection
from biothings_explorer.hint import Hint
ht = Hint()

## Functions that will be used
# Check for every intermediate node type in Predict funciton
def predict_many(input_object, intermediate_node_list, output_type):
    df_list = []
    for inter in intermediate_node_list:
        try: 
            print("Intermediate Node type running:")
            print(inter)
            fc = FindConnection(input_obj=input_object, output_obj=output_type, intermediate_nodes=[inter])
            fc.connect(verbose=False)
            df = fc.display_table_view()
            rows = df.shape[0]
            if(rows > 0):
                df_list.append(df)
        except:
            print("FAILED")
    if(len(df_list) > 0):
        return pd.concat(df_list)
    else:
        return None
    
# all intermediate node types

node_type_list = (['Gene', 'SequenceVariant', 'ChemicalSubstance', 'Disease', 
                   'MolecularActivity', 'BiologicalProcess', 'CellularComponent', 
                   'Pathway', 'AnatomicalEntity', 'PhenotypicFeature'])

## 1.1 TMPRSS2 -> NAFAMOSTAT
### In what ways can the TMPRSS2 Gene be connected to NAFAMOSTAT?
### 1.1.1 TMPRSS2 -> NAFAMOSTAT Query 

In [43]:
tmprss2 = ht.query("TMPRSS2")['Gene'][0]

all_tmprss2_predictions = predict_many(tmprss2,node_type_list,'ChemicalSubstance')

Intermediate Node type running:
Gene
Intermediate Node type running:
SequenceVariant
Intermediate Node type running:
ChemicalSubstance
Intermediate Node type running:
Disease
API 8.7 pharos failed
API 8.4 pharos failed
API 8.15 pharos failed
API 8.1 pharos failed
API 8.14 pharos failed
API 8.12 pharos failed
API 8.2 pharos failed
API 8.6 pharos failed
API 8.13 pharos failed
API 8.10 pharos failed
API 8.3 pharos failed
API 8.5 pharos failed
API 8.11 pharos failed
API 8.8 pharos failed
API 8.9 pharos failed
API 8.17 pharos failed
API 8.22 pharos failed
API 8.16 pharos failed
API 8.18 pharos failed
API 8.21 pharos failed
API 8.23 pharos failed
API 8.20 pharos failed
API 8.25 pharos failed
API 8.19 pharos failed
API 8.36 pharos failed
API 8.31 pharos failed
API 8.32 pharos failed
API 8.33 pharos failed
API 8.26 pharos failed
API 8.35 pharos failed
API 8.29 pharos failed
API 8.34 pharos failed
API 8.37 pharos failed
API 8.24 pharos failed
API 8.27 pharos failed
API 8.30 pharos failed
API 8.

### 1.1.2 TMPRSS2 -> NAFAMOSTAT Results table 

In [44]:
## FUT 175 (Futhan) is another name for Nafamostat 
all_tmprss2_predictions[(all_tmprss2_predictions["output_name"] == "FUT 175")]

input input_type                      pred1 pred1_source  \
30984  TMPRSS2       Gene  physically_interacts_with         None   
2405   TMPRSS2       Gene                 related_to      scibite   
2406   TMPRSS2       Gene                 related_to     scigraph   
2407   TMPRSS2       Gene  physically_interacts_with     drugbank   
43018  TMPRSS2       Gene                 related_to      scibite   
43019  TMPRSS2       Gene                 related_to          CTD   

                         pred1_api  \
30984                  BioLink API   
2405    Automat CORD19 Scibite API   
2406   Automat CORD19 Scigraph API   
2407               MyChem.info API   
43018   Automat CORD19 Scibite API   
43019                      CTD API   

                                       pred1_pubmed         node1_type  \
30984                                      32296183               Gene   
2405                                           None  ChemicalSubstance   
2406                                           None  ChemicalSubstance   
2407                                           None  ChemicalSubstance   
43018                                          None            Disease   
43019  24027332|24227843|27550352|27733646|28778717            Disease   

                           node1_name             node1_id  \
30984                             TNF        NCBIGene:7124   
2405                         CAMOSTAT        name:CAMOSTAT   
2406                         CAMOSTAT        name:CAMOSTAT   
2407                         CAMOSTAT        name:CAMOSTAT   
43018  CORONAVINAE INFECTIOUS DISEASE  MONDO:MONDO:0005719   
43019  CORONAVINAE INFECTIOUS DISEASE  MONDO:MONDO:0005719   

                           pred2 pred2_source                   pred2_api  \
30984  physically_interacts_with         None                   DGIdb API   
2405        negatively_regulates       SEMMED         SEMMED Chemical API   
2406        negatively_regulates       SEMMED         SEMMED Chemical API   
2407        negatively_regulates       SEMMED         SEMMED Chemical API   
43018                 related_to      scibite  Automat CORD19 Scibite API   
43019                 related_to      scibite  Automat CORD19 Scibite API   

      pred2_pubmed        output_type output_name     output_id  
30984         None  ChemicalSubstance     FUT 175  name:FUT 175  
2405       9322090  ChemicalSubstance     FUT 175  name:FUT 175  
2406       9322090  ChemicalSubstance     FUT 175  name:FUT 175  
2407       9322090  ChemicalSubstance     FUT 175  name:FUT 175  
43018         None  ChemicalSubstance     FUT 175  name:FUT 175  
43019         None  ChemicalSubstance     FUT 175  name:FUT 175

## 1.2 Question: Is the only way to get from TMPRSS2 to Blood Coagulation through Nafamostat (Fut 175)? 

In [48]:
tmprss2_to_blood_coagulation = predict_many(tmprss2,node_type_list,'BiologicalProcess')

Intermediate Node type running:
Gene
Intermediate Node type running:
SequenceVariant
Intermediate Node type running:
ChemicalSubstance
Intermediate Node type running:
Disease
Intermediate Node type running:
MolecularActivity
Intermediate Node type running:
BiologicalProcess
Intermediate Node type running:
CellularComponent
Intermediate Node type running:
Pathway
Intermediate Node type running:
AnatomicalEntity
Intermediate Node type running:
PhenotypicFeature


### 1.2.1 TMPRSS2 to Blood Coagulation through any node type:

In [50]:
tmprss2_to_blood_coagulation[tmprss2_to_blood_coagulation['output_name'].str.contains('COAGULATION', regex=False)]

input input_type                      pred1  \
4950   TMPRSS2       Gene  physically_interacts_with   
8807   TMPRSS2       Gene  physically_interacts_with   
8808   TMPRSS2       Gene  physically_interacts_with   
8809   TMPRSS2       Gene  physically_interacts_with   
8810   TMPRSS2       Gene  physically_interacts_with   
364    TMPRSS2       Gene                 related_to   
1117   TMPRSS2       Gene                 related_to   
1118   TMPRSS2       Gene                 related_to   
1119   TMPRSS2       Gene                 related_to   
1120   TMPRSS2       Gene                 related_to   
5379   TMPRSS2       Gene                     causes   
5380   TMPRSS2       Gene                 related_to   
5381   TMPRSS2       Gene                     causes   
5382   TMPRSS2       Gene                 related_to   
5383   TMPRSS2       Gene                 related_to   
5384   TMPRSS2       Gene                 related_to   
5385   TMPRSS2       Gene                     treats   
5386   TMPRSS2       Gene                 related_to   
5387   TMPRSS2       Gene                    affects   
5388   TMPRSS2       Gene                     treats   
5389   TMPRSS2       Gene                 related_to   
5390   TMPRSS2       Gene                    affects   
5391   TMPRSS2       Gene                 related_to   
5392   TMPRSS2       Gene                 related_to   
5393   TMPRSS2       Gene                 related_to   
5394   TMPRSS2       Gene                 related_to   
5395   TMPRSS2       Gene                 related_to   
5396   TMPRSS2       Gene                 related_to   
5397   TMPRSS2       Gene                 related_to   
5398   TMPRSS2       Gene                 related_to   
5399   TMPRSS2       Gene                 related_to   
5400   TMPRSS2       Gene                 related_to   
5401   TMPRSS2       Gene                 related_to   
11905  TMPRSS2       Gene                 related_to   
12746  TMPRSS2       Gene                 related_to   
12747  TMPRSS2       Gene                 related_to   
12748  TMPRSS2       Gene                 related_to   
12749  TMPRSS2       Gene                 related_to   
907    TMPRSS2       Gene     functional_association   
97     TMPRSS2       Gene     functional_association   
98     TMPRSS2       Gene     functional_association   
330    TMPRSS2       Gene                 related_to   

                          pred1_source                    pred1_api  \
4950                            SEMMED              SEMMED Gene API   
8807                            SEMMED              SEMMED Gene API   
8808                              None                  BioLink API   
8809                              None                  BioLink API   
8810                              None                  BioLink API   
364                           scigraph  Automat CORD19 Scigraph API   
1117                           scibite   Automat CORD19 Scibite API   
1118                          scigraph  Automat CORD19 Scigraph API   
1119                           scibite   Automat CORD19 Scibite API   
1120                          scigraph  Automat CORD19 Scigraph API   
5379                            SEMMED              SEMMED Gene API   
5380                            SEMMED              SEMMED Gene API   
5381                            SEMMED              SEMMED Gene API   
5382                            SEMMED              SEMMED Gene API   
5383                            SEMMED              SEMMED Gene API   
5384                            SEMMED              SEMMED Gene API   
5385                            SEMMED              SEMMED Gene API   
5386                            SEMMED              SEMMED Gene API   
5387                            SEMMED              SEMMED Gene API   
5388                            SEMMED              SEMMED Gene API   
5389                            SEMMED              SEMMED Gene API   
5390                            SEMMED

### 1.2.2 TMPRSS2 to Blood Coagulation through any ChemicalSubstance type node:

In [55]:
tmprss2_to_blood_coagulation[(tmprss2_to_blood_coagulation['node1_type'] == "ChemicalSubstance") & (tmprss2_to_blood_coagulation['output_name'].str.contains('COAGULATION', regex=False))]

input input_type       pred1 pred1_source  \
364   TMPRSS2       Gene  related_to     scigraph   
1117  TMPRSS2       Gene  related_to      scibite   
1118  TMPRSS2       Gene  related_to     scigraph   
1119  TMPRSS2       Gene  related_to      scibite   
1120  TMPRSS2       Gene  related_to     scigraph   

                        pred1_api pred1_pubmed         node1_type  \
364   Automat CORD19 Scigraph API         None  ChemicalSubstance   
1117   Automat CORD19 Scibite API         None  ChemicalSubstance   
1118  Automat CORD19 Scigraph API         None  ChemicalSubstance   
1119   Automat CORD19 Scibite API         None  ChemicalSubstance   
1120  Automat CORD19 Scigraph API         None  ChemicalSubstance   

       node1_name           node1_id       pred2  \
364   CHEBI:35222  CHEBI:CHEBI:35222  related_to   
1117      FUT 175       name:FUT 175     affects   
1118      FUT 175       name:FUT 175     affects   
1119      FUT 175       name:FUT 175    disrupts   
1120      FUT 175       name:FUT 175    disrupts   

                         pred2_source            pred2_api pred2_pubmed  \
364   Translator Text Mining Provider    CORD Chemical API         None   
1117                           SEMMED  SEMMED Chemical API      3443413   
1118                           SEMMED  SEMMED Chemical API      3443413   
1119                           SEMMED  SEMMED Chemical API      3443413   
1120                           SEMMED  SEMMED Chemical API      3443413   

            output_type        output_name               output_id  
364   BiologicalProcess        COAGULATION           GO:GO:0050817  
1117  BiologicalProcess  BLOOD COAGULATION  name:BLOOD COAGULATION  
1118  BiologicalProcess  BLOOD COAGULATION  name:BLOOD COAGULATION  
1119  BiologicalProcess  BLOOD COAGULATION  name:BLOOD COAGULATION  
1120  BiologicalProcess  BLOOD COAGULATION  name:BLOOD COAGULATION

1.2 NOTES: Tables show that TMPRSS2 -> coagulation can also be found through 'CHEBI:35222' Chemical Substance, as well as through many other node types that are not drugs

## 1.3 TMPRSS2 -> Camostat
### 1.3.1 In what ways can the TMPRSS2 Gene be connected to Camostat?

In [56]:
all_tmprss2_predictions[all_tmprss2_predictions["output_name"] == "CAMOSTAT"]

input input_type                      pred1 pred1_source  \
3172   TMPRSS2       Gene       negatively_regulates       SEMMED   
3173   TMPRSS2       Gene  physically_interacts_with       SEMMED   
3174   TMPRSS2       Gene  physically_interacts_with         None   
3175   TMPRSS2       Gene  physically_interacts_with         None   
3176   TMPRSS2       Gene  physically_interacts_with         None   
42864  TMPRSS2       Gene                 related_to      scibite   
42865  TMPRSS2       Gene                 related_to     scigraph   
42866  TMPRSS2       Gene                 related_to          CTD   
42867  TMPRSS2       Gene                 related_to      DISEASE   
42868  TMPRSS2       Gene                 related_to      scibite   
42869  TMPRSS2       Gene                 related_to          CTD   
42870  TMPRSS2       Gene                 related_to      scibite   
42871  TMPRSS2       Gene                 related_to     scigraph   
42872  TMPRSS2       Gene                 related_to      scibite   
42873  TMPRSS2       Gene                 related_to      scibite   
42874  TMPRSS2       Gene                 related_to      scibite   
42875  TMPRSS2       Gene                 related_to     scigraph   

                         pred1_api  \
3172               SEMMED Gene API   
3173               SEMMED Gene API   
3174                   BioLink API   
3175                   BioLink API   
3176                   BioLink API   
42864   Automat CORD19 Scibite API   
42865  Automat CORD19 Scigraph API   
42866                      CTD API   
42867                 DISEASES API   
42868   Automat CORD19 Scibite API   
42869                      CTD API   
42870   Automat CORD19 Scibite API   
42871  Automat CORD19 Scigraph API   
42872   Automat CORD19 Scibite API   
42873   Automat CORD19 Scibite API   
42874   Automat CORD19 Scibite API   
42875  Automat CORD19 Scigraph API   

                                       pred1_pubmed node1_type  \
3172                                       25221645       Gene   
3173                                       25520876       Gene   
3174                                       21068237       Gene   
3175                                           None       Gene   
3176                                           None       Gene   
42864                                          None    Disease   
42865                                          None    Disease   
42866                    22496216|24227843|30626688    Disease   
42867                                          None    Disease   
42868                                          None    Disease   
42869  24027332|24227843|27550352|27733646|28778717    Disease   
42870                                          None    Disease   
42871                                          None    Disease   
42872                                          None    Disease   
42873                                          None    Disease   
42874                                          None    Disease   
42875                                          None    Disease   

                                       node1_name             node1_id  \
3172                                     C1710301        UMLS:C1710301   
3173                                     C1710301        UMLS:C1710301   
3174                                         ACE2       NCBIGene:59272   
3175                                      TMPRSS4       NCBIGene:56649   
3176                                      TMPRSS4       NCBIGene:56649   
42864           SEVERE ACUTE RESPIRATORY SYNDROME  MONDO:MONDO:0005091   
42865           SEVERE ACUTE RESPIRATORY SYNDROME  MONDO:MONDO:0005091   
42866           SEVERE ACUTE RESPIRATORY SYNDROME  MONDO:MONDO:0005091   
42867           SEVERE ACUTE RESPIRATORY SYNDROME  MONDO:MONDO:0005091   
42868              CORONAVINAE INFECTIOUS DISEASE  MONDO:MONDO:0005719   
42869              CORONAVINAE INFECTIOUS DISEASE  MONDO:MONDO:0005719   
4287

# 2 CAMOSTAT Related Queries

## 2.1 Camostat to Protease Activity 
### 2.1.1 Camostat to Protease Activity Query 

In [4]:
camostat = ht.query("CAMOSTAT")["ChemicalSubstance"][0]

In [6]:
## while not used (yet), serine_protease_activity shows BiologicalProcess nodes could be used as the output type
serine_protease_activity = ht.query("protease")['BiologicalProcess'][0]
serine_protease_activity    

{'name': 'serine protease activity',
 'UMLS': 'C1150239',
 'primary': {'identifier': 'UMLS',
  'cls': 'BiologicalProcess',
  'value': 'C1150239'},
 'display': 'UMLS(C1150239) name(serine protease activity)',
 'type': 'BiologicalProcess'}

In [9]:
camastat_BP_query_results = predict_many(camostat, node_type_list, 'BiologicalProcess')

Intermediate Node type running:
Gene
Intermediate Node type running:
SequenceVariant
Intermediate Node type running:
ChemicalSubstance
Intermediate Node type running:
Disease
API 8.1 pharos failed
Intermediate Node type running:
MolecularActivity
Intermediate Node type running:
BiologicalProcess
Intermediate Node type running:
CellularComponent
Intermediate Node type running:
Pathway
Intermediate Node type running:
AnatomicalEntity
Intermediate Node type running:
PhenotypicFeature


### 2.1.2 Camostat to Protease Activity Results

In [13]:
camastat_BP_query_results[camastat_BP_query_results["output_name"].str.contains('PROTEASE', regex=False)]

input         input_type                      pred1 pred1_source  \
3888  CAMOSTAT  ChemicalSubstance                 related_to     scigraph   
6508  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
6509  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
1197  CAMOSTAT  ChemicalSubstance  physically_interacts_with       SEMMED   
1198  CAMOSTAT  ChemicalSubstance       negatively_regulates       SEMMED   
1199  CAMOSTAT  ChemicalSubstance  physically_interacts_with       SEMMED   
1200  CAMOSTAT  ChemicalSubstance       negatively_regulates       SEMMED   
1257  CAMOSTAT  ChemicalSubstance  physically_interacts_with       SEMMED   
1258  CAMOSTAT  ChemicalSubstance       negatively_regulates       SEMMED   
946   CAMOSTAT  ChemicalSubstance                     treats       SEMMED   
947   CAMOSTAT  ChemicalSubstance                 related_to      scibite   

                        pred1_api     pred1_pubmed         node1_type  \
3888  Automat CORD19 Scigraph API             None               Gene   
6508                      CTD API         20023007               Gene   
6509                      CTD API         27773435               Gene   
1197          SEMMED Chemical API         21541544  ChemicalSubstance   
1198          SEMMED Chemical API         21528279  ChemicalSubstance   
1199          SEMMED Chemical API         21541544  ChemicalSubstance   
1200          SEMMED Chemical API         21528279  ChemicalSubstance   
1257          SEMMED Chemical API         21541544  ChemicalSubstance   
1258          SEMMED Chemical API         21528279  ChemicalSubstance   
946           SEMMED Chemical API  15796165,229707            Disease   
947    Automat CORD19 Scibite API             None            Disease   

                        node1_name                           node1_id  \
3888                          DPP4                      NCBIGene:1803   
6508                           TNF                      NCBIGene:7124   
6509                           TNF                      NCBIGene:7124   
1197  SERINE PROTEINASE INHIBITORS  name:SERINE PROTEINASE INHIBITORS   
1198  SERINE PROTEINASE INHIBITORS  name:SERINE PROTEINASE INHIBITORS   
1199  SERINE PROTEINASE INHIBITORS  name:SERINE PROTEINASE INHIBITORS   
1200  SERINE PROTEINASE INHIBITORS  name:SERINE PROTEINASE INHIBITORS   
1257  SERINE PROTEINASE INHIBITORS  name:SERINE PROTEINASE INHIBITORS   
1258  SERINE PROTEINASE INHIBITORS  name:SERINE PROTEINASE INHIBITORS   
946                   PANCREATITIS                MONDO:MONDO:0004982   
947                   PANCREATITIS                MONDO:MONDO:0004982   

           pred2 pred2_source            pred2_api               pred2_pubmed  \
3888    disrupts       SEMMED      SEMMED Gene API                   20155839   
6508    disrupts       SEMMED      SEMMED Gene API                   27491710   
6509    disrupts       SEMMED      SEMMED Gene API                   27491710   
1197     affects       SEMMED  SEMMED Chemical API  20536561,23624075,7665170   
1198     affects       SEMMED  SEMMED Chemical API  20536561,23624075,7665170   
1199    disrupts       SEMMED  SEMMED Chemical API           22143069,2884198   
1200    disrupts       SEMMED  SEMMED Chemical API           22143069,2884198   
1257    disrupts       SEMMED  SEMMED Chemical API                    2992505   
1258    disrupts       SEMMED  SEMMED Chemical API                    2992505   
946   related_to       SEMMED   SEMMED Disease API                     159836   
947   related_to       SEMMED   SEMMED Disease API                     159836   

            output_type                      output_name  \
3888  BiologicalProcess         SERINE PROTEASE ACTIVITY   
6508  BiologicalProcess  SUMO-SPECIFIC PROTEASE ACTIVITY   
6509  BiologicalProcess  SUMO-SPECIFIC PROTEASE ACTIVITY   
1197  BiologicalProcess         SERINE PROTEASE ACTIVITY   
1198  BiologicalProcess         SERINE PROTEASE ACTIVITY 

## 2.2 Camostat to Protease Related Genes
### 2.2.1 Camostat to Protease Related Genes Query

In [14]:
camastat_Gene_query_results = predict_many(camostat, node_type_list, 'Gene')

Intermediate Node type running:
Gene
Intermediate Node type running:
SequenceVariant
Intermediate Node type running:
ChemicalSubstance
Intermediate Node type running:
Disease
API 8.1 pharos failed
Intermediate Node type running:
MolecularActivity
Intermediate Node type running:
BiologicalProcess
Intermediate Node type running:
CellularComponent
Intermediate Node type running:
Pathway
Intermediate Node type running:
AnatomicalEntity
Intermediate Node type running:
PhenotypicFeature


In [15]:
camastat_Gene_query_results

input         input_type                 pred1 pred1_source  \
0     CAMOSTAT  ChemicalSubstance  negatively_regulates       SEMMED   
1     CAMOSTAT  ChemicalSubstance  negatively_regulates       SEMMED   
2     CAMOSTAT  ChemicalSubstance  negatively_regulates       SEMMED   
3     CAMOSTAT  ChemicalSubstance            related_to          CTD   
4     CAMOSTAT  ChemicalSubstance            related_to          CTD   
...        ...                ...                   ...          ...   
8343  CAMOSTAT  ChemicalSubstance              disrupts       SEMMED   
8344  CAMOSTAT  ChemicalSubstance              disrupts       SEMMED   
8345  CAMOSTAT  ChemicalSubstance              disrupts       SEMMED   
8346  CAMOSTAT  ChemicalSubstance              disrupts       SEMMED   
8347  CAMOSTAT  ChemicalSubstance              disrupts       SEMMED   

                pred1_api       pred1_pubmed         node1_type  \
0     SEMMED Chemical API           11291936               Gene   
1     SEMMED Chemical API           11291936               Gene   
2     SEMMED Chemical API           11291936               Gene   
3                 CTD API  25766432,27773435               Gene   
4                 CTD API           26599507               Gene   
...                   ...                ...                ...   
8343  SEMMED Chemical API            6205521  BiologicalProcess   
8344  SEMMED Chemical API            6205521  BiologicalProcess   
8345  SEMMED Chemical API            6205521  BiologicalProcess   
8346  SEMMED Chemical API            6205521  BiologicalProcess   
8347  SEMMED Chemical API            6205521  BiologicalProcess   

            node1_name               node1_id                      pred2  \
0             C2985149          UMLS:C2985149       negatively_regulates   
1             C2985149          UMLS:C2985149  physically_interacts_with   
2             C2985149          UMLS:C2985149       positively_regulates   
3                 CCN2          NCBIGene:1490       negatively_regulates   
4                SMAD2          NCBIGene:4087       positively_regulates   
...                ...                    ...                        ...   
8343  TRYPSIN ACTIVITY  name:TRYPSIN ACTIVITY                   disrupts   
8344  TRYPSIN ACTIVITY  name:TRYPSIN ACTIVITY                   disrupts   
8345  TRYPSIN ACTIVITY  name:TRYPSIN ACTIVITY                   disrupts   
8346  TRYPSIN ACTIVITY  name:TRYPSIN ACTIVITY                    affects   
8347  TRYPSIN ACTIVITY  name:TRYPSIN ACTIVITY                   disrupts   

     pred2_source                      pred2_api  \
0          SEMMED                SEMMED Gene API   
1          SEMMED                SEMMED Gene API   
2          SEMMED                SEMMED Gene API   
3          SEMMED                SEMMED Gene API   
4          SEMMED                SEMMED Gene API   
...           ...                            ...   
8343       SEMMED  SEMMED Biological Process API   
8344       SEMMED  SEMMED Biological Process API   
8345       SEMMED  SEMMED Biological Process API   
8346       SEMMED  SEMMED Biological Process API   
8347       SEMMED  SEMMED Biological Process API   

                                      pred2_pubmed output_type output_name  \
0              11306691,12381730,12514171,18562499        Gene    C1705556   
1     10226149,10777731,10962016,11306691,12514171        Gene    C1705556   
2                                 1338049,15978904        Gene    C1705556   
3                                         24288211        Gene    C1705556   
4                                         27184078        Gene    C1705556   
...                                            ...         ...         ...   
8343                                      16339296        Gene    C0054756   
8344                                       1707846        Gene    C0078939   
8345                                      11853759        Gene    C0532449   
8346                 

In [26]:
## which fo the camastat_Gene_query_results have protease-related activity? 
gene_list = list(camastat_Gene_query_results["output_name"])
gene_list = list(dict.fromkeys(gene_list))
# print(gene_list)
print(len(gene_list))
## Would be nice if there was way to search UMLS id types.. ie UMLS:C0052502

11829


In [39]:
## list genes with "protease" in full name. this genes will be used to intersect the camastat_Gene_query_results
#### Need to come up with a quicker way to do this
gene_list2 = gene_list[9907:]
protease_related_genes_identified = []
index = 9907
for gene_name in gene_list2:
    if(index % 100 == 0):
#         print("Index")
#         print(index)
    current_query = ht.query(gene_name)
    if(len(current_query['Gene']) > 0):
        if(('PROTEASE' in current_query['Gene'][0]['name']) or ('protease' in current_query['Gene'][0]['name'])):
            print(gene_name)
            protease_related_genes_identified.append(gene_name)
    index = index + 1
    

In [38]:
protease_related_genes_identified = ['PRSS2','PRSS3','TMPRSS4','PRSS8','TMPRSS13','PRSS55','TMPRSS12','ST14','TMPRSS9','TMPRSS2','TMPRSS11D','TMPRSS11A','PRSS1','PRSS50','PRSS27','PRSS12','TMPRSS15','PRSS58','PRSS23','PRSS57']

### 2.2.2 Camostat to Protease Related Genes Results

In [40]:
camastat_Gene_query_results[camastat_Gene_query_results["output_name"].isin(protease_related_genes_identified)]

input         input_type                      pred1 pred1_source  \
2198  CAMOSTAT  ChemicalSubstance  physically_interacts_with     drugbank   
2199  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2200  CAMOSTAT  ChemicalSubstance  physically_interacts_with         None   
2201  CAMOSTAT  ChemicalSubstance                 related_to       pharos   
2202  CAMOSTAT  ChemicalSubstance  physically_interacts_with     drugbank   
...        ...                ...                        ...          ...   
5488  CAMOSTAT  ChemicalSubstance                   disrupts       SEMMED   
6567  CAMOSTAT  ChemicalSubstance                   disrupts       SEMMED   
7046  CAMOSTAT  ChemicalSubstance                   disrupts       SEMMED   
7811  CAMOSTAT  ChemicalSubstance                   disrupts       SEMMED   
7813  CAMOSTAT  ChemicalSubstance                   disrupts       SEMMED   

                pred1_api    pred1_pubmed         node1_type node1_name  \
2198      MyChem.info API  143965,7629790               Gene      PRSS1   
2199      MyChem.info API            None               Gene      PRSS1   
2200            DGIdb API            None               Gene      PRSS1   
2201   Automat PHAROS API            None               Gene      PRSS1   
2202      MyChem.info API  143965,7629790               Gene      PRSS1   
...                   ...             ...                ...        ...   
5488  SEMMED Chemical API        21541544  BiologicalProcess     GROWTH   
6567  SEMMED Chemical API        21541544  BiologicalProcess     GROWTH   
7046  SEMMED Chemical API        21541544  BiologicalProcess     GROWTH   
7811  SEMMED Chemical API        21541544  BiologicalProcess     GROWTH   
7813  SEMMED Chemical API        21541544  BiologicalProcess     GROWTH   

           node1_id                      pred2 pred2_source  \
2198  NCBIGene:5644  physically_interacts_with         None   
2199  NCBIGene:5644  physically_interacts_with         None   
2200  NCBIGene:5644  physically_interacts_with         None   
2201  NCBIGene:5644  physically_interacts_with         None   
2202  NCBIGene:5644  physically_interacts_with         None   
...             ...                        ...          ...   
5488    name:GROWTH                   disrupts       SEMMED   
6567    name:GROWTH                   disrupts       SEMMED   
7046    name:GROWTH                   disrupts       SEMMED   
7811    name:GROWTH                    affects       SEMMED   
7813    name:GROWTH                    affects       SEMMED   

                          pred2_api                         pred2_pubmed  \
2198                    BioLink API                                 None   
2199                    BioLink API                                 None   
2200                    BioLink API                                 None   
2201                    BioLink API                                 None   
2202                    BioLink API                                 None   
...                             ...                                  ...   
5488  SEMMED Biological Process API                             26993610   
6567  SEMMED Biological Process API                             20303767   
7046  SEMMED Biological Process API                             11708800   
7811  SEMMED Biological Process API  14654438,17892472,18389234,27190239   
7813  SEMMED Biological Process API                             27983914   

     output_type output_name       output_id  
2198        Gene       PRSS2   NCBIGene:5645  
2199        Gene       PRSS2   NCBIGene:5645  
2200        Gene       PRSS2   NCBIGene:5645  
2201        Gene       PRSS2   NCBIGene:5645  
2202        Gene       PRSS3   NCBIGene:5646  
...          ...         ...             ...  
5488        Gene     TMPRSS4  NCBIGene:56649  
6567        Gene   TMPRSS11D   NCBIGene:9407  
7046        Gene        ST14   NCBIGene:6768  
7811        Gene       PRSS1   NCBIGene:564

## 2.3 Camostat to Coagulation
### 2.3.1 Camostat to Coagulation Results

In [51]:
camastat_caog_BP_query_results = camastat_BP_query_results[camastat_BP_query_results["output_name"].str.contains("COAGULATION")]
camastat_caog_BP_query_results

input         input_type                      pred1 pred1_source  \
2717  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2718  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2719  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
2720  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2721  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
2722  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2723  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
2724  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2751  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2752  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2753  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2754  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
2922  CAMOSTAT  ChemicalSubstance  physically_interacts_with  drugcentral   
2923  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
2924  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
2925  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
2926  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
4860  CAMOSTAT  ChemicalSubstance                 related_to          CTD   
20    CAMOSTAT  ChemicalSubstance  physically_interacts_with       SEMMED   
21    CAMOSTAT  ChemicalSubstance  physically_interacts_with       SEMMED   
22    CAMOSTAT  ChemicalSubstance  physically_interacts_with       SEMMED   
23    CAMOSTAT  ChemicalSubstance       negatively_regulates       SEMMED   
24    CAMOSTAT  ChemicalSubstance  physically_interacts_with       SEMMED   
25    CAMOSTAT  ChemicalSubstance       negatively_regulates       SEMMED   
26    CAMOSTAT  ChemicalSubstance  physically_interacts_with       SEMMED   
27    CAMOSTAT  ChemicalSubstance       negatively_regulates       SEMMED   
28    CAMOSTAT  ChemicalSubstance       positively_regulates       SEMMED   
29    CAMOSTAT  ChemicalSubstance       negatively_regulates       SEMMED   
30    CAMOSTAT  ChemicalSubstance       negatively_regulates       SEMMED   
3     CAMOSTAT  ChemicalSubstance                     treats       SEMMED   
4     CAMOSTAT  ChemicalSubstance                   disrupts       SEMMED   
5     CAMOSTAT  ChemicalSubstance                     treats       SEMMED   
6     CAMOSTAT  ChemicalSubstance                 related_to      scibite   
11    CAMOSTAT  ChemicalSubstance                   disrupts       SEMMED   

                       pred1_api     pred1_pubmed         node1_type  \
2717             MyChem.info API             None               Gene   
2718             MyChem.info API             None               Gene   
2719                     CTD API         27773435               Gene   
2720             MyChem.info API             None               Gene   
2721                     CTD API         27773435               Gene   
2722             MyChem.info API             None               Gene   
2723                     CTD API         25766432               Gene   
2724             MyChem.info API             None               Gene   
2751             MyChem.info API             None               Gene   
2752             MyChem.info API             None               Gene   
2753             MyChem.info API             None               Gene   
2754                     CTD API         25766432               Gene   
2922             MyChem.info API             None               Gene   
2923                     CTD API         27773435               Gene   
2924                     CTD API         25766432               Gene   
2925                     CTD API         20023007               Gene   
2926                     CTD API         27773435               Gene   
4860             

In [52]:
camastat_caog_genes = camastat_caog_BP_query_results[camastat_caog_BP_query_results['node1_type'] == "Gene"]

In [53]:
camastat_caog_genes_names = list(camastat_caog_genes["node1_name"])

camastat_caog_genes_names = list(dict.fromkeys(camastat_caog_genes_names))

print("IDENTIFIED GENES")
print(camastat_caog_genes_names)


coag_protease_related_genes_identified = []
for gene_name in camastat_caog_genes_names :
    current_query = ht.query(gene_name)
    if(len(current_query['Gene']) > 0):
        print(current_query['Gene'][0])
        if(('PROTEASE' in current_query['Gene'][0]['name']) or ('protease' in current_query['Gene'][0]['name'])):
            print(gene_name)
            coag_protease_related_genes_identified.append(gene_name)
            
## NONE OF THE GENES HAD "PROTEASE IN NAME"

IDENTIFIED GENES
['F2', 'PLG', 'HGFAC', 'COL1A1', 'SERPINE1', 'TNF']
{'NCBIGene': '2147', 'name': 'coagulation factor II, thrombin', 'SYMBOL': 'F2', 'UMLS': 'C1414504', 'HGNC': '3535', 'UNIPROTKB': 'P00734', 'ENSEMBL': 'ENSG00000180210', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '2147'}, 'display': 'NCBIGene(2147) ENSEMBL(ENSG00000180210) HGNC(3535) UMLS(C1414504) UNIPROTKB(P00734) SYMBOL(F2)', 'type': 'Gene'}
{'NCBIGene': '5340', 'name': 'plasminogen', 'SYMBOL': 'PLG', 'UMLS': 'C1367553', 'HGNC': '9071', 'UNIPROTKB': 'P00747', 'ENSEMBL': 'ENSG00000122194', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '5340'}, 'display': 'NCBIGene(5340) ENSEMBL(ENSG00000122194) HGNC(9071) UMLS(C1367553) UNIPROTKB(P00747) SYMBOL(PLG)', 'type': 'Gene'}
{'NCBIGene': '3083', 'name': 'HGF activator', 'SYMBOL': 'HGFAC', 'UMLS': 'C1415534', 'HGNC': '4894', 'UNIPROTKB': 'Q04756', 'ENSEMBL': 'ENSG00000109758', 'primary': {'identifier': 'NCBIGene', 'cls': 'Gene', 'value': '30